### PR TITLE
Allow component definitions to set/overwrite parameters of their dependencies.

### PR DESCRIPTION
**Pull request by Viech**

### CODE REVIEW

File: CBSE.py [hunk 1]
```diff
@@ -229,12 +233,34 @@ def gather_components(self, components, mandatory):
...
             for param in component.param_list:
                 if not param.name in self.params[component.name]:
+                    # Use the parameters own default value, if it has one, as fallback
                     if param.default != None:
                         self.params[component.name][param.name] = param.default
-                    else:
+
+                    # Look for a value given by another component that depends on the current one
+                    dependency_sets_parameter = False
+                    for depender in self.components:
```
Comment (Kangz) [inline CBSE.py]: nit: dependant

Comment (Kangz) [inline CBSE.py]: nit dependant or dependency

**Comment by Kangz** [comment]
Changes look good, worst case complexity looks scary, we'll see how bad it is in practice.

**Comment by Viech** [comment]
@Kangz should be my last commit now. I'd work on running time in case it becomes an issue.

**Comment by Kangz** [comment]
LGTM

**Pull request merged**